api/src/routes/schedule.routes.ts

```diff
@@ -1,4 +1,100 @@
-// import express from 'express';
+import express from 'express';
+import { ScheduleController } from '../controllers/schedule.controller';
+
+const router = express.Router();
+const scheduleController: ScheduleController = new ScheduleController();
+
+
+/**
+ * admin triggers build schedule
+ * Make schedule by running the algorithms (currently just creates a mock schedule)
+ * @param {*} req
+ * @param {*} res
+ * @return {*}
+ */
+const create = async (req, res) => {
+    try
+    {
+        const response = await scheduleController.create();
+        res.status(200).send("Created schedule.");
+    } catch (err)
+    {
+        res.status(401).send({message: err});
+    }
+}
+router.get('/create', create);
+
+
+
+/**
+ * Admin list entire schedule
+ * Get the entire schedule that has been created from the database
+ * @param {*} req
+ * @param {*} res
+ * @return {*} ISchedule[]
+*/
+const list = async (req, res) => {
+    if(!req.headers.authorization)
+    {
+        res.status(400).send({ message: "This endpoint requires authorization header."});
+    }
+
+    const token = req.headers.authorization;
+
+    try
+    {
+        const response = await scheduleController.list(token);
+        res.status(200).send(response);
+    } catch (err)
+    {
+        res.status(401).send("Error: " + err);
+    }
+}
+router.get('/list', list);
+
+
+/** teacher: get my schedule
+ *  * @param {*} req
+ * @param {*} res
+ * @return {*} ISchedule[]
+*/
+const my = async (req, res) => {
+    if(!req.headers.authorization)
+    {
+        res.status(400).send({ message: "This endpoint requires authorization header."});
+    }
+
+    const token = req.headers.authorization;
+
+    try
+    {
+        const response = await scheduleController.my(token);
```

Comment:
Instead of creating a new function called `my' which collects user data on its own. Could it init a user.controller to get the user data and then separately have a function in schedule that gets all schedules based from a user email or id. That way we maintain any data invloved in parsing JWT is maintained inside the auth or user controller

Reply:
Not sure if I understood it corrctly; I'm using user.controller to get user data now, and I don't pass authtoken to my (or other functions in the schedule controller). Let me know if I should change something.